django/contrib/auth/hashers.py

```diff
@@ -478,16 +478,18 @@ def must_update(self, encoded):
         return int(decoded_entry['work_factor']) != self.rounds
 
     def harden_runtime(self, password, encoded):
-        decoded_entry = self.decode(encoded)
-        data = "$".join(('', decoded_entry['algostr'],
-                        decoded_entry['work_factor'],
-                        decoded_entry['salt']))
+        _, data = encoded.split('$', 1)
+        salt = data[:29]  # Length of the salt in bcrypt.
+        rounds = data.split('$')[2]
         # work factor is logarithmic, adding one doubles the load.
-        diff = 2**(self.rounds - int(decoded_entry['work_factor'])) - 1
+        diff = 2**(self.rounds - int(rounds)) - 1
         while diff > 0:
-            self.encode(password, data.encode('ascii'))
+            self.encode(password, salt.encode('ascii'))
             diff -= 1
 
+        hash = hashlib.sha1((salt + password).encode()).hexdigest()
```

Comment:
I think this got copied from somewhere else by mistake and should be dropped (including the following line)

Reply:
Right you are. Fixed. I squashed a few commits down as well to clean up the sprawl